### PR TITLE
[vds/combiner] Teach VDS combiner how to handle custom call fields

### PR DESCRIPTION
**Pull request by chrisvittal**
Add the call_fields argument.

Much like refrence entries, when passing a VDS to the combiner, we check it's call fields to ensure that there is a match between the passed argument and the VDS's call fields.

### CODE REVIEW

File: hail/python/hail/vds/combiner/variant_dataset_combiner.py [hunk 1]
```diff
@@ -156,7 +157,7 @@ class VariantDatasetCombiner:  # pylint: disable=too-many-instance-attributes
         for all GVCF files. Finer partitioning yields more parallelism but less work per task.
     gvcf_info_to_keep : :class:`list` of :class:`str` or :obj:`None`
         GVCF ``INFO`` fields to keep in the ``gvcf_info`` entry field. By default, all fields are
```
Comment (danking) [inline hail/python/hail/vds/combiner/variant_dataset_combiner.py:159]: need to remove the `are` on the previous line

File: hail/python/hail/vds/combiner/variant_dataset_combiner.py [hunk 2]
```diff
@@ -701,6 +709,16 @@ def maybe_load_from_saved_path(save_path: str) -> Optional[VariantDatasetCombine
...
         gvcf_reference_entry_fields_to_keep = ref_entry_tmp
 
+        all_entry_types = chain(vds.reference_data._type.entry.items(),
+                                vds.variant_data._type.entry.items())
+        call_fields_tmp = {name for name, typ in all_entry_types if typ == hl.tcall} - {'LGT', 'GT'}
+        if 'LPGT' in call_fields_tmp:
+            call_fields_tmp = (call_fields_tmp - {'LPGT'}) | {'PGT'}
+        if set(call_fields) != call_fields_tmp:
+            warning("Mismatch between 'call_fields' and VDS call fields. "
+                    "Overwriting with types from supplied VDS")
```
Comment (danking) [inline hail/python/hail/vds/combiner/variant_dataset_combiner.py:719]: Include the mismatch. In particular, the user should know the `call_fields` argument but might not know what's in the dataset.


What do you mean by "overwriting with types ..."? It's not the types that are taken from the VDS but the call fields that are taken from it, right?

File: hail/python/hail/vds/combiner/variant_dataset_combiner.py [hunk 3]
```diff
@@ -701,6 +709,16 @@ def maybe_load_from_saved_path(save_path: str) -> Optional[VariantDatasetCombine
                     "entry types. Overwriting with types from supplied VDS.")
         gvcf_reference_entry_fields_to_keep = ref_entry_tmp
 
+        all_entry_types = chain(vds.reference_data._type.entry.items(),
+                                vds.variant_data._type.entry.items())
+        call_fields_tmp = {name for name, typ in all_entry_types if typ == hl.tcall} - {'LGT', 'GT'}
```
Comment (danking) [inline hail/python/hail/vds/combiner/variant_dataset_combiner.py:714]: These are the call fields from the vds, right? Let's use a descriptive name. Maybe `vds_call_fields`?

**Review by danking — CHANGES_REQUESTED** [review]

**Comment by danking** [comment]
Mostly me not knowing how this code works, so, my apologies for slowing you down.

File: hail/python/hail/vds/combiner/variant_dataset_combiner.py (hunk 2 above)
Comment (chrisvittal) [reply hail/python/hail/vds/combiner/variant_dataset_combiner.py:719]: Will do. Also will clarify.

File: hail/python/hail/vds/combiner/variant_dataset_combiner.py (hunk 3 above)
Comment (chrisvittal) [reply hail/python/hail/vds/combiner/variant_dataset_combiner.py:714]: 👍

File: hail/python/hail/vds/combiner/variant_dataset_combiner.py [hunk 4]
```diff
@@ -701,6 +709,16 @@ def maybe_load_from_saved_path(save_path: str) -> Optional[VariantDatasetCombine
...
+        if 'LPGT' in call_fields_tmp:
+            call_fields_tmp = (call_fields_tmp - {'LPGT'}) | {'PGT'}
```
[the remark this answers is not in the document]
Comment (chrisvittal) [reply hail/python/hail/vds/combiner/variant_dataset_combiner.py:716]: We transform certain fields from their 'normal' gvcf versions to the VDS 'local' ones, PGT is one of them and is also the only VCF field we parse as a call field by default.

**Review by danking — APPROVED** [review]
Great!

**Pull request merged**